src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/general/MissingEqualsAndHashCodeDiagnosticRule.java

```diff
@@ -67,16 +68,16 @@ private boolean hasValidHashCodeMethod(TypeDeclaration<?> typeDeclaration) {
 
     private boolean isValidEqualsMethod(MethodDeclaration method) {
         return method.getNameAsString().equals("equals")
-                && method.isAnnotationPresent("Override")
-                && method.getParameters().size() == 1
-                && method.getParameters().get(0).getType().asString().equals("Object")
-                && method.getType().asString().equals("boolean");
+            && method.getParameters().size() == 1
+            && method.isAnnotationPresent("Override")
```

Comment:
The order is switched because `isAnnotationPresent` check is more expensive than checking the size? If so, we should do this below as well in `isValidHashCodeMethod()`.